### ISSUE TITLE
[Feature request] Alter square deadzone on controller to a circular one

### ISSUE THREAD

**Issue by sologuitarplaying** [nipkownix/re4_tweaks#198]
### Describe the feature you'd like to see added

This is the last major problem with aiming weapons on a controller. In the gamecube version of the game it's equally easy to make fine diagonal adjustments while aiming as it is vertical or horizontal, but not so in the modern ports. This change would make controller aiming feel much smoother.

**Comment by LethalPlacebo** [comment]
Part of this is referenced in issue #118 and it's worth mentioning: the deadzones in modern ports of RE4 are not square, but cross. RE4 tweaks already includes an option to make the overall deadzone smaller, but Steam's controller input customization also allows to fine-tune the cross deadzone.

**Comment by sologuitarplaying** [comment]
> Part of this is referenced in issue #118 and it's worth mentioning: the deadzones in modern ports of RE4 are not square, but cross. RE4 tweaks already includes an option to make the overall deadzone smaller, but Steam's controller input customization also allows to fine-tune the cross deadzone.

Thanks for the info! Sounds like the real challenge is getting the cross deadzone mitigated for aiming, while preserving it for movement.

**Comment by LethalPlacebo** [comment]
> Thanks for the info! (corrected title) Sounds like the real challenge is getting the cross deadzone mitigated for aiming, while preserving it for movement. Could you elaborate on fine-tuning the cross deadzone via steam input? The anti-deadzone setting is effective, but diagonal aiming adjustments are still less responsive than vertical or horizontal ones, as far as I can tell.

Hey, sorry for the late response, the anti-deadzone setting in Steam input was exactly what I was talking about. It's worth noting that I use Type III controls, the ones that put camera and aiming on the right stick. Using an anti-deadzone output of 0.275, I was able to get diagonal movements feeling really good.

Are you sure diagonals are less responsive than horizontal or vertical? For me they feel just as responsive, the problem was that the effective area of diagonals was tiny because of the deadzone. Using RE4 Tweaks + Steam input as set above, the diagonals feel just as responsive to me.

**Comment by LethalPlacebo** [comment]
> Just became aware that changing the deadzone shape setting to cross also applies to the anti-deadzone, and it's much better with that in order. Still a slight preference for cardinal directions though, because even shrunk, it remains cross shaped. I think you'll agree if you directly compare diagonally aiming as slowly as possible on the gamecube version to the steam version.
> 
> I suppose the issue is very minor with type 3 controls, but still stands for types 1 and 2, due to aiming and movement sharing the same deadzone shape.

Yes, I do believe you are correct that diagonals still feel better on the GameCube version (which you can even feel playing it on Dolphin) than on my current setup. I still think my current controls feel great, but yeah it'd be nice to get them exactly as they were originally because of how good it feels. Another post mentioned how the deadzone changes while aiming in the GC version in #118, I believe it was a point made by 80TC.

**Comment by sologuitarplaying** [comment]
Figured out a solution to the problem of the shared deadzone with type 1 and 2 controls. In steam input there's an option called "mode shift" that allows you to assign an additional function to a button or joystick. Using it, you can configure an alternate mode for the left joystick, where it still outputs as left joystick, but with a different deadzone from the primary mode. And if you set this mode to activate when you hold down right trigger, you have something very close to GC behavior. 

One quirk with this is that if you're holding a direction on the stick when you press the aim button, Leon's aim will jump to reflect the sudden change in deadzone. Interestingly, I checked the GC version, and the same thing happens there.

**Comment by LethalPlacebo** [comment]
> Figured out a solution to the problem of the shared deadzone with type 1 and 2 controls. In steam input there's an option called "mode shift" that allows you to assign an additional function to a button or joystick. Using it, you can configure an alternate mode for the left joystick, where it still outputs as left joystick, but with a different deadzone from the primary mode. And if you set this mode to activate when you hold down right trigger, you have something very close to GC behavior.
> 
> One quirk with this is that if you're holding a direction on the stick when you press the aim button, Leon's aim will jump to reflect the sudden change in deadzone. Interestingly, I checked the GC version, and the same thing happens there.

Yeah, this was the same thing that 80TC recommended. If is weird that they got rid of the idea of separate deadzones for moving and aiming in the modern ports, especially considering that Leon's actual movement is always digital.

**Comment by sologuitarplaying** [comment]
My guess is they didn't know any better. The GC version seemingly has no aiming deadzone set by the game at all and relies on the controller's self-calibration. Modern controllers can't do that, so it's likely the porting team just went with what was already in the game.

**Comment by LethalPlacebo** [comment]
> My guess is they didn't know any better. The GC version seemingly has no aiming deadzone set by the game at all and relies on the controller's self-calibration when a weapon is raised. Modern controllers can't do that, so it's likely the porting team just went with what was already in the game, the one intended for 8 way movement.

Ah, I didn't even realize those differences between the GC controller and modern controllers were a thing!

**Comment by pas-de-2** [comment]
In addition to the aiming discrepancies, something I've noticed from testing back and forth between UHD and Dolphin is that the diagonals for movement on Gamecube are very small. The deadzone itself is smaller than re4_tweaks with the xinput override set to 0.4, but much more of the stick's angle is devoted to cross directional movement This makes Gamecube movement feel both more responsive and easier to control than PC

If you try and lower re4_tweaks' deadzone slider to match Gamecube's sensitivity, Leon develops a problem with moving in a straight line, because shrinking the deadzone is also modifying the activation angles for some reason. To see what I mean, hold up on the left stick and then slowly ride the gate until you hit the activation point for diagonal movement, and watch how this point changes as you mess with the xinput override.
